src/lib/db.js

```diff
@@ -126,39 +133,28 @@ export function infoPagesQuery(slugs, columns) {
   });
 }
 
-export function articleQuery(slugs, columns) {
-  // parameters are both expected to be arrays
-  // first one with article slugs to fetch
-  // second one which columns to fetch from the articles table
-  return new Promise((resolve) => {
-    // we join with articles table to find slug, and always return slug
-    let processedColumns = columns.map((col) => {
-      // make it compatible for the sql query
-      if (col === 'category') {
-        return 'categories.slug as category';
-      } if (col === 'published_at') {
-        return 'gazelle_published_at as published_at';
-      }
-      return col;
-    });
-    // Put slug there so we know what we fetched
-    // Use concat to make a copy, if you just push
-    // it will change pathSet in the falcorPath
-    // as objects are passed by reference
-    processedColumns = processedColumns.concat(['articles.slug as slug']);
-    database.select(...processedColumns)
+/**
+ * Fetches direct meta data of articles from the articles database table
+ * @param {string[]} slugs - Array of slugs of articles to fetch
+ * @param {string[]} columns - Which columns of the articles table to fetch
+ * @returns {Promise<Object[]>}
+ */
+export async function articleQuery(slugs, columns) {
+  const processedColumns = columns.map((col) => {
+    // make it compatible for the sql query
+    if (col === 'category') {
+      return 'categories.slug as category';
+    }
+    return `articles.${col}`;
+  });
+  // In order to be able to identify the rows we get back we need to include the slug
+  if (!processedColumns.includes('articles.slug')) {
+    processedColumns.push('articles.slug');
+  }
+  return await database.select(...processedColumns)
```

Comment:
Yes, if an error is thrown Falcor will return all the objects requested but instead of actual values it will have the error thrown as the value.

That said, I do really want to implement some proper error logging, and when I'm done with this part I'm actually planning on spending some time on giving us some better helper functions such as setting up an error logger we can call (and for example use try catch around this), and I also want to setup the Falcor HOC so we can get rid of the Falcor Controller